src/FINDOLOGIC/Api/Responses/Xml21/Xml21Response.php

```diff
@@ -68,26 +63,38 @@ protected function buildResponseElementInstances($response)
         $this->promotion = $this->getPromotionFromResponse($xmlResponse);
         $this->results = new Results($xmlResponse->results[0]);
 
-        foreach ($xmlResponse->products->children() as $product) {
-            $productId = ResponseHelper::getStringProperty($product->attributes(), 'id', true);
-            // Set product ids as keys for the products.
-            $this->products[$productId] = new Product($product);
+        try {
+            foreach ($xmlResponse->products->children() as $product) {
+                $productId = ResponseHelper::getStringProperty($product->attributes(), 'id', true);
+                // Set product ids as keys for the products.
+                $this->products[$productId] = new Product($product);
+            }
+        } catch (Exception $ignored) {
+            // Do nothing if an exception is thrown as it does not need to be executed on error
         }
 
-        foreach ($xmlResponse->filters->main->children() as $filter) {
-            $filterName =  ResponseHelper::getStringProperty($filter, 'name');
-            // Set filter names as keys for the filters.
-            $this->mainFilters[$filterName] = new Filter($filter);
-            $this->hasMainFilters = true;
-            $this->mainFilterCount++;
+        try {
+            foreach ($xmlResponse->filters->main->children() as $filter) {
+                $filterName = ResponseHelper::getStringProperty($filter, 'name');
+                // Set filter names as keys for the filters.
+                $this->mainFilters[$filterName] = new Filter($filter);
+                $this->hasMainFilters = true;
+                $this->mainFilterCount++;
+            }
+        } catch (Exception $ignored) {
+            // Do nothing if an exception is thrown as it does not need to be executed on error
         }
 
-        foreach ($xmlResponse->filters->other->children() as $filter) {
-            $filterName =  ResponseHelper::getStringProperty($filter, 'name');
-            // Set filter names as keys for the filters.
-            $this->otherFilters[$filterName] = new Filter($filter);
-            $this->hasOtherFilters = true;
-            $this->otherFilterCount++;
+        try {
```

Comment:
Having just a try/catch catching all exceptions is not a clean solution.
Check if `$xmlResponse->filters->other` and `$xmlResponse->filters->other->children()` exist and are not empty. If they have a value and are not empty, do the foreach.